test/test_routes.py

```diff
@@ -862,3 +862,50 @@ def test_component_server_endpoints(connect):
             },
         )
         assert fail_req.status_code == 404
+
+
+@pytest.mark.parametrize(
+    "request_url, route_path, root_path, expected_root_url",
+    [
+        ("http://localhost:7860/", "/", None, "http://localhost:7860"),
```

Comment:
I think adding a a test where route_path ends in `/` could be useful. I know we have the `/` test here though.